docs/markdown_code_symlinks.py

```diff
@@ -113,65 +157,80 @@ def find_links(cls):
-        if src_rdoc not in self.mapping['docs2code']:
-            # Don't do any rewriting on non-linked markdown.
-            url = ourl
-
-        # Is the destination also inside docs?
-        elif dst_rsrc not in self.mapping['code2docs']:
-            # Return a path to the GitHub repo.
-            url = "{}/blob/master/{}".format(
-                self.config['github_code_repo'], dst_rsrc)
+    @classmethod
+    def remove_extension(cls, path):
+        return filename
+
+    # Overriden method to solve the cross-reference link
+    def resolve_xref(
+            self, env, fromdocname, builder, typ, target, node, contnode):
+        if '#' in target:
+            todocname, targetid = target.split('#')
+        else:
+            todocname = target
+            targetid = ''
+
+        # Removing filename extension (e.g. contributing.md -> contributing)
+        todocname, _ = os.path.splitext(self.mapping['code2docs'][todocname])
+
+        newnode = make_refnode(
+            builder, fromdocname, todocname, targetid, contnode[0])
+
+        return newnode
+
+    def resolve_any_xref(
+            self, env, fromdocname, builder, target, node, contnode):
+        res = self.resolve_xref(
+            env, fromdocname, builder, 'xref', target, node, contnode)
+        return [('prjxray:xref', res)]
+
+
+class LinkParser(parser.CommonMarkParser, object):
+    def visit_link(self, mdnode):
+        ref_node = nodes.reference()
+
+        destination = mdnode.destination
+
+        ref_node.line = self._get_line(mdnode)
+        if mdnode.title:
+            ref_node['title'] = mdnode.title
+
+        url_check = urlparse(destination)
+        # If there's not a url scheme (e.g. 'https' for 'https:...' links),
+        # or there is a scheme but it's not in the list of known_url_schemes,
+        # then assume it's a cross-reference and pass it to Sphinx as an `:any:` ref.
+        known_url_schemes = self.config.get('known_url_schemes')
+        if known_url_schemes:
+            scheme_known = url_check.scheme in known_url_schemes
         else:
-            url = os.path.relpath(
-                os.path.join(
-                    self.docs_root_dir, self.mapping['code2docs'][dst_rsrc]),
-                start=src_dir)
-            base_url, ext = os.path.splitext(url)
-            assert ext in (".md",
-                           ".markdown"), ("Unknown extension {}".format(ext))
-            url = "{}.html".format(base_url)
-
-        print("---")
-        print(ourl)
-        print(url)
-        print
-        return url
+            scheme_known = bool(url_check.scheme)
+
+        is_dest_xref = url_check.fragment and destination[:1] != '#'
+
+        ref_node['refuri'] = destination
+        next_node = ref_node
+
+        # Adds pending-xref wrapper to unsolvable cross-references
+        if (is_dest_xref or not url_check.fragment) and not scheme_known:
+            wrap_node = addnodes.pending_xref(
```

Comment:
Do links to headers work?

Reply:
By headers you mean links to other html pages without the use of fragments (e.g. a xref from CONTRIBUTING.md to SUPPORT.md)?
In this case yes, they do work